src/main/java/com/odysseusinc/arachne/portal/api/v1/dto/converters/study/BaseStudyToStudyDTOConverter.java

```diff
@@ -82,9 +83,17 @@ public DTO convert(final S source) {
                 )
         );
 
+        final Tenant studyTenant = source.getTenant();
+        
         for (final StudyDataSourceLink studyDataSourceLink : foundLinks) {
+            
             final DataSourceDTO dataSourceDTO = conversionService.convert(studyDataSourceLink,
                     DataSourceDTO.class);
+            
+            if (!studyDataSourceLink.getDataSource().getTenants().contains(studyTenant)) {
```

Comment:
What about performance and amount of queries with such approach?

Reply:
Contains will be comparing ids, so entities will not be initialized